File: JavaScript/tests/problem063_test.js
const chai = require('chai')
const assert = chai.assert
const problem063 = require('../problem063');
const canFormWord = problem063.canFormWord;

// Implement kind of a factory function
function getTestMatrix() {
    return [['F', 'A', 'C', 'I'],
            ['O', 'B', 'Q', 'P'],
            ['A', 'N', 'O', 'B'],
            ['M', 'A', 'S', 'S']]
}

describe('Problem 63', function(){
  it("canFormWord finds horizontal matches", function() {
    assert.isTrue(canFormWord(getTestMatrix(), "mass"))
  });

  it("canFormWord finds vertical matches", function() {
    assert.isTrue(canFormWord(getTestMatrix(), "foam"))
  });

});

File: Ruby/tests/problem063_test.rb
require 'minitest/autorun'
require_relative '../problem063'

class Problem063Test < Minitest::Test
  def setup
    @test_matrix = [['F', 'A', 'C', 'I'],
                    ['O', 'B', 'Q', 'P'],
                    ['A', 'N', 'O', 'B'],
                    ['M', 'A', 'S', 'S']]
  end

  def test_can_find_words_horizontally
    assert Problem063.can_form_word?(@test_matrix, 'mass')
  end

  def test_can_find_vertical_words
    assert Problem063.can_form_word?(@test_matrix, 'foam')
  end
end
